src/azure-cli/azure/cli/command_modules/synapse/operations/sqlpoolblobauditingpolicy.py

```diff
@@ -58,7 +95,16 @@ def sqlpool_blob_auditing_policy_update(
                 "FAILED_DATABASE_AUTHENTICATION_GROUP",
                 "BATCH_COMPLETED_GROUP"]
 
-    return instance
+
+def _validate_audit_policy_arguments(state=None, storage_account=None, storage_endpoint=None,
+                                     storage_account_access_key=None, retention_days=None):
+    blob_storage_arguments_provided = any(
+        [storage_account, storage_endpoint, storage_account_access_key, retention_days])
+    if not state and not blob_storage_arguments_provided:
+        raise CLIError('Either state or blob storage arguments are missing')
+
+    if retention_days is not None and (not retention_days.isdigit() or int(retention_days) <= 0):
+        raise CLIError('retention-days must be a positive number greater than zero')
```

Comment:
such validation would be recommended to add as validator and other commands/parameters could utilize the function.
see https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/resource/commands.py#L240

Reply:
> such validation would be recommended to add as validator and other commands/parameters could utilize the function.
> see https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/storage/_params.py#L351

Hi @Juliehzl , thanks for your comment, I have removed the validation for retention_days. I agree with you, but per my understanding, the validator need to bind with one parameter which means one validator for one parameter, how should we validate multiple parameters in validator?

Reply:
Hi @Juliehzl , for this comments, could you please give some suggestions? Thanks

Reply:
I have moved this function to _validators.py and add validator in command level. Thanks for this comments, it is my first time to know that we can add validator in command level.